### PR TITLE
Modal Page

### PR DESCRIPTION
**Pull request by JiangTianXiang**
-Completed Modal for home page
-Refactored App.js to use Hooks
-Refactored DisplayImage, ImageAndUser component to fit for modal
-Resolved previous pull request conversations
-Minor refactor for all pages

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -10,39 +10,35 @@ import SearchCollectionResult from "pages/SearchCollectionResult";
...
+          <NavBar handleTheme={setTheme} on={theme} />
+          <Switch>
+            <Route
+              exact
+              path="/"
+              component={Home}
+            ></Route>
+            <Route path="/user/:name" component={User}></Route>
+            <Route
+              path="/search/photos/:input"
```
Comment (yoloOnTheBattlefield) [inline src/App.js]: `path="/search/photos/:searchTerm"` sounds better

File: src/components/Modal/index.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  const [saved, setSaved] = useState(false);
+
+  const handleSave = () => {
+    saved ? removeFavoriteImage(data) : saveFavoriteImage(data);
+    setSaved(!saved);
+  };
+
+  useEffect(() => {
+    if (data) {
+      const founded = localStorage.getItem(data.id);
```
Comment (yoloOnTheBattlefield) [inline src/components/Modal/index.js]: found?

File: src/pages/Home/index.js [hunk 3]
```diff
@@ -31,11 +31,12 @@ export default class Home extends React.Component {
 
   componentDidMount() {
     this.getData();
+    document.body.style.overflow = "unset";
```
Comment (yoloOnTheBattlefield) [inline src/pages/Home/index.js]: why is this in here?

Comment (JiangTianXiang) [reply src/pages/Home/index.js]: Leftover from the previous version. Deleted

File: src/components/Modal/ModalImage/ModalImage.styles.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import styled, { css } from "styled-components";
+
+export const Container = styled.div`
+  position: relative;
+  ${(props) => props.imageContainerCSS};
```
Comment (yoloOnTheBattlefield) [inline src/components/Modal/ModalImage/ModalImage.styles.js]: why? this seems a bit mind bending?

can you do 
${props => props.boolean? css``}
?

File: src/components/Modal/index.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useState } from "react";
+import ReactDOM from "react-dom";
+import favoriteIcon from "utils/resources/Iconly-Broken-Star.svg";
+import savedFavoriteIcon from "utils/resources/Iconly-Filled-Star.svg";
+import likeIcon from "utils/resources/Iconly-Broken-Heart.svg";
+import closeIcon from "utils/resources/Icon-metro-cross.svg";
+import ModalImage from "components/Modal/ModalImage";
+import Author from "components/Author";
```
Comment (yoloOnTheBattlefield) [inline src/components/Modal/index.js]: import {Author, ModalImage} from 'components'

same with the rest

File: src/pages/SearchPhotoResult/index.js [hunk 6]
```diff
@@ -29,7 +29,7 @@ export default class SearchPhotoResult extends React.Component {
 
   getData = async () => {
     try {
-      const searchInput = this.props.match.params.input;
+      const searchInput = this.props.match.params.searchTerm;
       console.log(this.state.page);
```
Comment (yoloOnTheBattlefield) [inline src/pages/SearchPhotoResult/index.js]: meh

**Review by yoloOnTheBattlefield — APPROVED** [review]

**Pull request merged**